smart-contracts/contracts/RelayBridge.sol

```diff
@@ -77,27 +90,36 @@ contract RelayBridge is IRelayBridge {
       );
   }
 
-  /// @notice Bridge tokens from the L2 to the L1
+  /// @notice Bridges tokens from the L2 to the L1
+  /// @param amount Amount of tokens to bridge
+  /// @param recipient Address that will receive the bridged tokens
+  /// @param l1Asset Address of the L1 asset to bridge
+  /// @param l1Gas Gas limit for the L1 transaction
+  /// @param extraData Extra data to pass to the bridge proxy
   function bridge(
     uint256 amount,
     address recipient,
     address l1Asset,
-    uint256 l1Gas
+    uint256 l1Gas,
+    bytes calldata extraData
   ) external payable returns (uint256 nonce) {
-    // Associate the withdrawal to a unique id
-    nonce = transferNonce++;
-
-    // Encode the data for the cross-chain message
-    // No need to pass the ASSET since the bridge and the pool are ASSET-specific
-    bytes memory data = abi.encode(nonce, recipient, amount, block.timestamp);
+    BridgeTransaction memory transaction = BridgeTransaction({
+      amount: amount,
+      recipient: recipient,
+      l1Asset: l1Asset,
+      l1Gas: l1Gas,
+      extraData: extraData,
+      nonce: transferNonce++, // Associate the withdrawal to a unique id
```

Comment:
```suggestion
      nonce: nonce, // Associate the withdrawal to a unique id
```

Reply:
Well we need to increment `transferNonce` :)

Reply:
yes but it needs to be done before (see https://github.com/relayprotocol/relay-vaults/pull/485#discussion_r2140027928). If not the data/txParams will have an empty none as it has not been incremented yet when we do `abi.encode`